src/utils/broker/saleStatus.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    BrokerConstantsType,
+    ConfigurationType,
+    SaleInfoType,
+    blocksToTimeFormat,
+  } from '@poppyseed/lastic-sdk'
+  
+  enum TypeOfChain {
+    PARA = 'PARA',
+    RELAY = 'RELAY',
+    LOCAL = 'LOCAL',
+  }
+  
+  enum StatusCode {
+    Interlude = 'interlude',
+    LeadIn = 'leadIn',
+    Purchase = 'purchase',
+    Ended = 'ended',
+  }
+  
+  interface StatusInfo {
+    statusTitle: string
+    statusMessage: string
+  }
+  
+  const statusInfoMap: Record<StatusCode, StatusInfo> = {
+    [StatusCode.Interlude]: {
+      statusTitle: 'Interlude Period',
+      statusMessage: 'Time to renew your core!',
+    },
+    [StatusCode.LeadIn]: {
+      statusTitle: 'Lead-in Period',
+      statusMessage:
+        'Sales have started we are now in the lead-in period. The price is linearly decreasing with each block.',
+    },
+    [StatusCode.Purchase]: {
+      statusTitle: 'Purchase Period',
+      statusMessage: 'Sale is in the purchase period.',
+    },
+    [StatusCode.Ended]: {
+      statusTitle: 'Sale Ended',
+      statusMessage: 'The sale has ended.',
+    },
+  }
+  
+  // Non-exported util functions for saleStatus
+
+
+  function getSaleEnds(
+    saleInfo: SaleInfoType,
+    config: ConfigurationType,
+    constant: BrokerConstantsType,
+    typeOfChain: TypeOfChain = TypeOfChain.PARA,
+  ): number {
+    const divideBy2OrNot: 1 | 2 = typeOfChain === 'PARA' ? 2 : 1
+    return (
+      saleInfo.saleStart +
+      (config.regionLength * constant.timeslicePeriod) / divideBy2OrNot -
+      config.interludeLength
```

Comment:
The problematic code is this one, somehow this is probably not correctly defined, since every time when we think the sale ends it actually doesn't yet, you can still make sales, there must be some factor that is not taken into account into the equation.

Reply:
got it, ill make a follow up PR